Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Booking.hs

```diff
@@ -17,37 +17,93 @@ module Storage.Queries.Booking where
 
 import Domain.Types.Booking
 import Domain.Types.DriverQuote (DriverQuote)
+import Domain.Types.Location
+import qualified Domain.Types.LocationMapping as DLocationMapping
 import Domain.Types.Merchant
 import Domain.Types.RiderDetails (RiderDetails)
 import qualified Domain.Types.SearchRequest as DSR
 import Kernel.Prelude
 import Kernel.Storage.Esqueleto as Esq hiding (findById, isNothing)
+import Kernel.Types.Error
 import Kernel.Types.Id
 import Kernel.Types.Time
+import Kernel.Utils.Common
 import qualified Storage.Queries.DriverQuote as QDQuote
+import qualified Storage.Queries.FareParameters as QFareParameters
 import Storage.Queries.FullEntityBuilders
+import qualified Storage.Queries.LocationMapping as QLocationMapping
 import Storage.Tabular.Booking
 import Storage.Tabular.DriverQuote as DriverQuote
 
--- fareParams already created with driverQuote
+bookingTableToBookingConverter :: (Transactionable m) => BookingTable -> m Booking
+bookingTableToBookingConverter BookingTable {..} = do
+  locationMappings <- QLocationMapping.findByTagId id.getId
+  let toLocation = map (.location) (filter (\locationMapping -> locationMapping.order /= 0) locationMappings)
+  fromLocation <- sourceLocationFinder locationMappings & fromMaybeM (InternalError "from location is missing")
+  fareParams' <- QFareParameters.findById fareParametersId
+  fareParams <- fareParams' & fromMaybeM (InternalError "fare params is missing")
+  return
+    Booking
+      { id = Id id.getId,
+        ..
+      }
+
+sourceLocationFinder :: [DLocationMapping.LocationMapping] -> Maybe Location
+sourceLocationFinder locationMappings = do
+  let orderZeroMappings = filter (\locationMapping -> locationMapping.order == 0) locationMappings
+  if null orderZeroMappings
+    then Nothing
+    else do
+      let source = head orderZeroMappings
+      Just $ source.location
+
+bookingToBookingTableConverter :: Booking -> BookingTable
+bookingToBookingTableConverter Booking {..} =
+  BookingTable
+    { id = Id id.getId,
+      fareParametersId = fareParams.id,
+      ..
+    }
+
 create :: Booking -> SqlDB ()
-create dBooking =
-  withFullEntity dBooking $ \(booking, fromLoc, toLoc, _fareParams) -> do
-    Esq.create' fromLoc
-    Esq.create' toLoc
-    Esq.create' booking
-
-findById :: Transactionable m => Id Booking -> m (Maybe Booking)
-findById bookingId = buildDType $ do
-  res <-
-    Esq.findOne' $ do
-      rb <- from $ table @BookingT
-      where_ $ rb ^. BookingTId ==. val (toKey bookingId)
-      pure rb
-  join <$> mapM buildFullBooking res
-
-findBySearchReq :: (Transactionable m) => Id DSR.SearchRequest -> m (Maybe Booking)
-findBySearchReq searchReqId = buildDType $ do
+create booking = do
+  let tablex = bookingToBookingTableConverter booking
+  createBTable tablex
+
+createBTable :: BookingTable -> SqlDB ()
+createBTable = Esq.create
```

Comment:
Instead of doing all this, you can use `insertSelect` and insert only specific columns (in ride too)

Reply:
Can't use `insertSelect` because we don't want `fromLocation` and `toLocation` columns, but we need them in Domain.Types.Booking

Reply:
This is the exact reason why you should use it. If you don't want any column in table, don't use it in `insertSelect`